src/main/java/com/iexec/worker/sms/SmsService.java

```diff
@@ -50,12 +51,13 @@ public SmsService(CredentialsService credentialsService, SmsClient smsClient) {
     public Optional<TaskSecrets> fetchTaskSecrets(WorkerpoolAuthorization workerpoolAuthorization) {
         String chainTaskId = workerpoolAuthorization.getChainTaskId();
         String authorization = getAuthorizationString(workerpoolAuthorization);
-        ResponseEntity<SmsSecretResponse> response = smsClient.getUnTeeSecrets(authorization, workerpoolAuthorization);
-        if (!response.getStatusCode().is2xxSuccessful()) {
+        SmsSecretResponse smsResponse;
+        try {
+            smsResponse = smsClient.getUnTeeSecrets(authorization, workerpoolAuthorization);
+        } catch (FeignException e) {
```

Comment:
I suggest to log the exception here.

Reply:
`log.error("Blabla [chainTaskId:{}]", chainTaskId, e);`
?

Reply:
Sorry, missed this, fixed in 47a38ba